diff --git a/client/mysqlbinlog.cc b/client/mysqlbinlog.cc
index 20b854f..c93c6d6 100644
--- a/client/mysqlbinlog.cc
+++ b/client/mysqlbinlog.cc
@@ -783,6 +783,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
   print_event_info->short_form= short_form;
   Exit_status retval= OK_CONTINUE;
   IO_CACHE *const head= &print_event_info->head_cache;
+  IO_CACHE *const body= &print_event_info->body_cache;
 
   /*
     Format events are not concerned by --offset and such, we always need to
@@ -873,7 +874,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
         print_event_info->skipped_event_in_transaction= false;
 
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       break;
       
@@ -931,7 +932,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
       */
       {
       ce->print(result_file, print_event_info, TRUE);
-        if (head->error == -1)
+        if (head->error == -1 || body->error == -1)
           goto err;
       }
       // If this binlog is not 3.23 ; why this test??
@@ -955,7 +956,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
         output of Append_block_log_event::print is only a comment.
       */
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       if ((retval= load_processor.process((Append_block_log_event*) ev)) !=
           OK_CONTINUE)
@@ -965,7 +966,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
     case EXEC_LOAD_EVENT:
     {
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       Execute_load_log_event *exv= (Execute_load_log_event*)ev;
       Create_file_log_event *ce= load_processor.grab_event(exv->file_id);
@@ -984,7 +985,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
 	ce->print(result_file, print_event_info, TRUE);
 	my_free((void*)ce->fname);
 	delete ce;
-        if (head->error == -1)
+        if (head->error == -1 || body->error == -1)
           goto err;
       }
       else
@@ -998,7 +999,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
       print_event_info->common_header_len=
         glob_description_event->common_header_len;
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       if (opt_remote_proto == BINLOG_LOCAL)
       {
@@ -1029,7 +1030,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
       break;
     case BEGIN_LOAD_QUERY_EVENT:
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       if ((retval= load_processor.process((Begin_load_query_log_event*) ev)) !=
           OK_CONTINUE)
@@ -1048,7 +1049,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
         {
           convert_path_to_forward_slashes(fname);
           exlq->print(result_file, print_event_info, fname);
-          if (head->error == -1)
+          if (head->error == -1 || body->error == -1)
           {
             if (fname)
               my_free(fname);
@@ -1191,7 +1192,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
       print_event_info->skipped_event_in_transaction= false;
 
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       break;
     }
@@ -1199,7 +1200,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
     {
       print_event_info->skipped_event_in_transaction= false;
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
       break;
     }
@@ -1213,7 +1214,7 @@ Exit_status process_event(PRINT_EVENT_INFO *print_event_info, Log_event *ev,
       /* fall through */
     default:
       ev->print(result_file, print_event_info);
-      if (head->error == -1)
+      if (head->error == -1 || body->error == -1)
         goto err;
     }
     /* Flush head cache to result_file for every event */
