diff --git a/include/my_sys.h b/include/my_sys.h
index ad1b645..b722954 100644
--- a/include/my_sys.h
+++ b/include/my_sys.h
@@ -234,13 +234,8 @@ extern void my_large_free(uchar *ptr);
 #define my_afree(PTR) my_free(PTR)
 #endif /* HAVE_ALLOCA */
 
-#ifndef errno				/* did we already get it? */
-#ifdef HAVE_ERRNO_AS_DEFINE
 #include <errno.h>			/* errno is a define */
-#else
-extern int errno;			/* declare errno */
-#endif
-#endif					/* #ifndef errno */
+
 extern char *home_dir;			/* Home directory for user */
 extern const char *my_progname;		/* program-name (printed in errors) */
 extern char curr_dir[];		/* Current directory for user */
